### PR TITLE
Fix large comment block headers so they don't cause phantom merge con…

### PR DESCRIPTION
**Pull request by jfy133**
…flicts

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Review by mashehu — APPROVED** [review]

**Comment by mashehu** [comment]
wait, you removed the harshil alignment 😱 
please put it back

**Comment by jfy133** [comment]
> wait, you removed the harshil alignment 😱 please put it back

Ben made me do it though :eyes: 

What do it Ldo :scream:

**Comment by mashehu** [comment]
I thought ben's secret thing followed harshil alignment. in any case in nf-core harshil alignment > ben

**Comment by jfy133** [comment]
No apparnetluy not everywhere...

I guess I'll go start from scratch then... :cry:

File: subworkflows/nf-core/utils_nextflow_pipeline/main.nf [hunk 1]
```diff
@@ -72,11 +71,11 @@ def getWorkflowVersion() {
...
-    def timestamp  = new java.util.Date().format( 'yyyy-MM-dd_HH-mm-ss')
-    def filename   = "params_${timestamp}.json"
-    def temp_pf    = new File(workflow.launchDir.toString(), ".${filename}")
-    def jsonStr    = groovy.json.JsonOutput.toJson(params)
-    temp_pf.text   = groovy.json.JsonOutput.prettyPrint(jsonStr)
+    def timestamp = new java.util.Date().format('yyyy-MM-dd_HH-mm-ss')
+    def filename = "params_${timestamp}.json"
+    def temp_pf = new File(workflow.launchDir.toString(), ".${filename}")
+    def jsonStr = groovy.json.JsonOutput.toJson(params)
+    temp_pf.text = groovy.json.JsonOutput.prettyPrint(jsonStr)
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nextflow_pipeline/main.nf:78]:
```suggestion
    def timestamp = new java.util.Date().format('yyyy-MM-dd_HH-mm-ss')
    def filename  = "params_${timestamp}.json"
    def temp_pf   = new File(workflow.launchDir.toString(), ".${filename}")
    def jsonStr   = groovy.json.JsonOutput.toJson(params)
    temp_pf.text  = groovy.json.JsonOutput.prettyPrint(jsonStr)
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 2]
```diff
@@ -112,46 +104,48 @@ def processVersionsFromYAML(yaml_file) {
...
-    yaml_file_text        += "plot_type: 'html'\n"
-    yaml_file_text        += "data: |\n"
-    yaml_file_text        += "${summary_section}"
+    def yaml_file_text = "id: '${workflow.manifest.name.replace('/', '-')}-summary'\n" as String
+    yaml_file_text += "description: ' - this information is collected when the pipeline is started.'\n"
+    yaml_file_text += "section_name: '${workflow.manifest.name} Workflow Summary'\n"
+    yaml_file_text += "section_href: 'https://github.com/${workflow.manifest.name}'\n"
+    yaml_file_text += "plot_type: 'html'\n"
+    yaml_file_text += "data: |\n"
+    yaml_file_text += "${summary_section}"
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:148]:
```suggestion
    def yaml_file_text = "id: '${workflow.manifest.name.replace('/', '-')}-summary'\n" as String
    yaml_file_text     += "description: ' - this information is collected when the pipeline is started.'\n"
    yaml_file_text     += "section_name: '${workflow.manifest.name} Workflow Summary'\n"
    yaml_file_text     += "section_href: 'https://github.com/${workflow.manifest.name}'\n"
    yaml_file_text     += "plot_type: 'html'\n"
    yaml_file_text     += "data: |\n"
    yaml_file_text     += "${summary_section}"
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 3]
```diff
@@ -190,63 +184,63 @@ def logColours(monochrome_logs=true) {
...
-    colorcodes['blink']      = monochrome_logs ? '' : "\033[5m"
-    colorcodes['reverse']    = monochrome_logs ? '' : "\033[7m"
-    colorcodes['hidden']     = monochrome_logs ? '' : "\033[8m"
+    colorcodes['reset'] = monochrome_logs ? '' : "[0m"
+    colorcodes['bold'] = monochrome_logs ? '' : "[1m"
+    colorcodes['dim'] = monochrome_logs ? '' : "[2m"
+    colorcodes['underlined'] = monochrome_logs ? '' : "[4m"
+    colorcodes['blink'] = monochrome_logs ? '' : "[5m"
+    colorcodes['reverse'] = monochrome_logs ? '' : "[7m"
+    colorcodes['hidden'] = monochrome_logs ? '' : "[8m"
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:193]:
```suggestion
    colorcodes['reset']      = monochrome_logs ? '' : "[0m"
    colorcodes['bold']       = monochrome_logs ? '' : "[1m"
    colorcodes['dim']        = monochrome_logs ? '' : "[2m"
    colorcodes['underlined'] = monochrome_logs ? '' : "[4m"
    colorcodes['blink']      = monochrome_logs ? '' : "[5m"
    colorcodes['reverse']    = monochrome_logs ? '' : "[7m"
    colorcodes['hidden']     = monochrome_logs ? '' : "[8m"
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 4]
```diff
@@ -190,63 +184,63 @@ def logColours(monochrome_logs=true) {
...
-    colorcodes['cyan']       = monochrome_logs ? '' : "\033[0;36m"
-    colorcodes['white']      = monochrome_logs ? '' : "\033[0;37m"
+    colorcodes['black'] = monochrome_logs ? '' : "[0;30m"
+    colorcodes['red'] = monochrome_logs ? '' : "[0;31m"
+    colorcodes['green'] = monochrome_logs ? '' : "[0;32m"
+    colorcodes['yellow'] = monochrome_logs ? '' : "[0;33m"
+    colorcodes['blue'] = monochrome_logs ? '' : "[0;34m"
+    colorcodes['purple'] = monochrome_logs ? '' : "[0;35m"
+    colorcodes['cyan'] = monochrome_logs ? '' : "[0;36m"
+    colorcodes['white'] = monochrome_logs ? '' : "[0;37m"
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:203]:
```suggestion
    colorcodes['black']  = monochrome_logs ? '' : "[0;30m"
    colorcodes['red']    = monochrome_logs ? '' : "[0;31m"
    colorcodes['green']  = monochrome_logs ? '' : "[0;32m"
    colorcodes['yellow'] = monochrome_logs ? '' : "[0;33m"
    colorcodes['blue']   = monochrome_logs ? '' : "[0;34m"
    colorcodes['purple'] = monochrome_logs ? '' : "[0;35m"
    colorcodes['cyan']   = monochrome_logs ? '' : "[0;36m"
    colorcodes['white']  = monochrome_logs ? '' : "[0;37m"
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 5]
```diff
@@ -190,63 +184,63 @@ def logColours(monochrome_logs=true) {
...
-    colorcodes['bcyan']      = monochrome_logs ? '' : "\033[1;36m"
-    colorcodes['bwhite']     = monochrome_logs ? '' : "\033[1;37m"
+    colorcodes['bblack'] = monochrome_logs ? '' : "[1;30m"
+    colorcodes['bred'] = monochrome_logs ? '' : "[1;31m"
+    colorcodes['bgreen'] = monochrome_logs ? '' : "[1;32m"
+    colorcodes['byellow'] = monochrome_logs ? '' : "[1;33m"
+    colorcodes['bblue'] = monochrome_logs ? '' : "[1;34m"
+    colorcodes['bpurple'] = monochrome_logs ? '' : "[1;35m"
+    colorcodes['bcyan'] = monochrome_logs ? '' : "[1;36m"
+    colorcodes['bwhite'] = monochrome_logs ? '' : "[1;37m"
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:213]:
```suggestion
    colorcodes['bblack']  = monochrome_logs ? '' : "[1;30m"
    colorcodes['bred']    = monochrome_logs ? '' : "[1;31m"
    colorcodes['bgreen']  = monochrome_logs ? '' : "[1;32m"
    colorcodes['byellow'] = monochrome_logs ? '' : "[1;33m"
    colorcodes['bblue']   = monochrome_logs ? '' : "[1;34m"
    colorcodes['bpurple'] = monochrome_logs ? '' : "[1;35m"
    colorcodes['bcyan']   = monochrome_logs ? '' : "[1;36m"
    colorcodes['bwhite']  = monochrome_logs ? '' : "[1;37m"
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 6]
```diff
@@ -190,63 +184,63 @@ def logColours(monochrome_logs=true) {
...
-    colorcodes['ucyan']      = monochrome_logs ? '' : "\033[4;36m"
-    colorcodes['uwhite']     = monochrome_logs ? '' : "\033[4;37m"
+    colorcodes['ublack'] = monochrome_logs ? '' : "[4;30m"
+    colorcodes['ured'] = monochrome_logs ? '' : "[4;31m"
+    colorcodes['ugreen'] = monochrome_logs ? '' : "[4;32m"
+    colorcodes['uyellow'] = monochrome_logs ? '' : "[4;33m"
+    colorcodes['ublue'] = monochrome_logs ? '' : "[4;34m"
+    colorcodes['upurple'] = monochrome_logs ? '' : "[4;35m"
+    colorcodes['ucyan'] = monochrome_logs ? '' : "[4;36m"
+    colorcodes['uwhite'] = monochrome_logs ? '' : "[4;37m"
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:223]:
```suggestion
    colorcodes['ublack']  = monochrome_logs ? '' : "[4;30m"
    colorcodes['ured']    = monochrome_logs ? '' : "[4;31m"
    colorcodes['ugreen']  = monochrome_logs ? '' : "[4;32m"
    colorcodes['uyellow'] = monochrome_logs ? '' : "[4;33m"
    colorcodes['ublue']   = monochrome_logs ? '' : "[4;34m"
    colorcodes['upurple'] = monochrome_logs ? '' : "[4;35m"
    colorcodes['ucyan']   = monochrome_logs ? '' : "[4;36m"
    colorcodes['uwhite']  = monochrome_logs ? '' : "[4;37m"
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 7]
```diff
@@ -190,63 +184,63 @@ def logColours(monochrome_logs=true) {
...
-    colorcodes['icyan']      = monochrome_logs ? '' : "\033[0;96m"
-    colorcodes['iwhite']     = monochrome_logs ? '' : "\033[0;97m"
+    colorcodes['iblack'] = monochrome_logs ? '' : "[0;90m"
+    colorcodes['ired'] = monochrome_logs ? '' : "[0;91m"
+    colorcodes['igreen'] = monochrome_logs ? '' : "[0;92m"
+    colorcodes['iyellow'] = monochrome_logs ? '' : "[0;93m"
+    colorcodes['iblue'] = monochrome_logs ? '' : "[0;94m"
+    colorcodes['ipurple'] = monochrome_logs ? '' : "[0;95m"
+    colorcodes['icyan'] = monochrome_logs ? '' : "[0;96m"
+    colorcodes['iwhite'] = monochrome_logs ? '' : "[0;97m"
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:233]:
```suggestion
    colorcodes['iblack']  = monochrome_logs ? '' : "[0;90m"
    colorcodes['ired']    = monochrome_logs ? '' : "[0;91m"
    colorcodes['igreen']  = monochrome_logs ? '' : "[0;92m"
    colorcodes['iyellow'] = monochrome_logs ? '' : "[0;93m"
    colorcodes['iblue']   = monochrome_logs ? '' : "[0;94m"
    colorcodes['ipurple'] = monochrome_logs ? '' : "[0;95m"
    colorcodes['icyan']   = monochrome_logs ? '' : "[0;96m"
    colorcodes['iwhite']  = monochrome_logs ? '' : "[0;97m"
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 8]
```diff
@@ -190,63 +184,63 @@ def logColours(monochrome_logs=true) {
...
-    colorcodes['bicyan']     = monochrome_logs ? '' : "\033[1;96m"
-    colorcodes['biwhite']    = monochrome_logs ? '' : "\033[1;97m"
+    colorcodes['biblack'] = monochrome_logs ? '' : "[1;90m"
+    colorcodes['bired'] = monochrome_logs ? '' : "[1;91m"
+    colorcodes['bigreen'] = monochrome_logs ? '' : "[1;92m"
+    colorcodes['biyellow'] = monochrome_logs ? '' : "[1;93m"
+    colorcodes['biblue'] = monochrome_logs ? '' : "[1;94m"
+    colorcodes['bipurple'] = monochrome_logs ? '' : "[1;95m"
+    colorcodes['bicyan'] = monochrome_logs ? '' : "[1;96m"
+    colorcodes['biwhite'] = monochrome_logs ? '' : "[1;97m"
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:243]:
```suggestion
    colorcodes['biblack']  = monochrome_logs ? '' : "[1;90m"
    colorcodes['bired']    = monochrome_logs ? '' : "[1;91m"
    colorcodes['bigreen']  = monochrome_logs ? '' : "[1;92m"
    colorcodes['biyellow'] = monochrome_logs ? '' : "[1;93m"
    colorcodes['biblue']   = monochrome_logs ? '' : "[1;94m"
    colorcodes['bipurple'] = monochrome_logs ? '' : "[1;95m"
    colorcodes['bicyan']   = monochrome_logs ? '' : "[1;96m"
    colorcodes['biwhite']  = monochrome_logs ? '' : "[1;97m"
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 9]
```diff
@@ -280,40 +275,49 @@ def attachMultiqcReport(multiqc_report) {
...
+        .sort()
+        .each { group ->
+            summary << summary_params[group]
+        }
 
     def misc_fields = [:]
-    misc_fields['Date Started']              = workflow.start
-    misc_fields['Date Completed']            = workflow.complete
+    misc_fields['Date Started'] = workflow.start
+    misc_fields['Date Completed'] = workflow.complete
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:293]:
```suggestion
    misc_fields['Date Started']              = workflow.start
    misc_fields['Date Completed']            = workflow.complete
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 10]
```diff
@@ -280,40 +275,49 @@ def attachMultiqcReport(multiqc_report) {
...
+        misc_fields['Pipeline repository Git URL'] = workflow.repository
+    }
+    if (workflow.commitId) {
+        misc_fields['Pipeline repository Git Commit'] = workflow.commitId
+    }
+    if (workflow.revision) {
+        misc_fields['Pipeline Git branch/tag'] = workflow.revision
+    }
+    misc_fields['Nextflow Version'] = workflow.nextflow.version
+    misc_fields['Nextflow Build'] = workflow.nextflow.build
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:306]:
```suggestion
    misc_fields['Pipeline script hash ID']   = workflow.scriptId
    if (workflow.repository) {
        misc_fields['Pipeline repository Git URL']    = workflow.repository
    }
    if (workflow.commitId) {
        misc_fields['Pipeline repository Git Commit'] = workflow.commitId
    }
    if (workflow.revision) {
        misc_fields['Pipeline Git branch/tag']        = workflow.revision
    }
    misc_fields['Nextflow Version']          = workflow.nextflow.version
    misc_fields['Nextflow Build']            = workflow.nextflow.build
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 11]
```diff
@@ -280,40 +275,49 @@ def attachMultiqcReport(multiqc_report) {
...
     misc_fields['Nextflow Compile Timestamp'] = workflow.nextflow.timestamp
 
     def email_fields = [:]
-    email_fields['version']      = getWorkflowVersion()
-    email_fields['runName']      = workflow.runName
-    email_fields['success']      = workflow.success
+    email_fields['version'] = getWorkflowVersion()
+    email_fields['runName'] = workflow.runName
+    email_fields['success'] = workflow.success
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:312]:
```suggestion
    email_fields['version']      = getWorkflowVersion()
    email_fields['runName']      = workflow.runName
    email_fields['success']      = workflow.success
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 12]
```diff
@@ -280,40 +275,49 @@ def attachMultiqcReport(multiqc_report) {
...
     email_fields['dateComplete'] = workflow.complete
-    email_fields['duration']     = workflow.duration
-    email_fields['exitStatus']   = workflow.exitStatus
+    email_fields['duration'] = workflow.duration
+    email_fields['exitStatus'] = workflow.exitStatus
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:315]:
```suggestion
    email_fields['duration']     = workflow.duration
    email_fields['exitStatus']   = workflow.exitStatus
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 13]
```diff
@@ -280,40 +275,49 @@ def attachMultiqcReport(multiqc_report) {
...
     email_fields['errorMessage'] = (workflow.errorMessage ?: 'None')
-    email_fields['errorReport']  = (workflow.errorReport ?: 'None')
-    email_fields['commandLine']  = workflow.commandLine
-    email_fields['projectDir']   = workflow.projectDir
-    email_fields['summary']      = summary << misc_fields
+    email_fields['errorReport'] = (workflow.errorReport ?: 'None')
+    email_fields['commandLine'] = workflow.commandLine
+    email_fields['projectDir'] = workflow.projectDir
+    email_fields['summary'] = summary << misc_fields
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:320]:
```suggestion
    email_fields['errorReport']  = (workflow.errorReport ?: 'None')
    email_fields['commandLine']  = workflow.commandLine
    email_fields['projectDir']   = workflow.projectDir
    email_fields['summary']      = summary << misc_fields
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 14]
```diff
@@ -325,51 +329,53 @@ def completionEmail(summary_params, email, email_on_fail, plaintext_email, outdi
     }
 
     // Render the TXT template
-    def engine       = new groovy.text.GStringTemplateEngine()
-    def tf           = new File("${workflow.projectDir}/assets/email_template.txt")
+    def engine = new groovy.text.GStringTemplateEngine()
+    def tf = new File("${workflow.projectDir}/assets/email_template.txt")
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:333]:
```suggestion
    def engine       = new groovy.text.GStringTemplateEngine()
    def tf           = new File("${workflow.projectDir}/assets/email_template.txt")
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 15]
```diff
@@ -325,51 +329,53 @@ def completionEmail(summary_params, email, email_on_fail, plaintext_email, outdi
...
     def txt_template = engine.createTemplate(tf).make(email_fields)
-    def email_txt    = txt_template.toString()
+    def email_txt = txt_template.toString()
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:335]:
```suggestion
    def email_txt    = txt_template.toString()
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 16]
```diff
@@ -325,51 +329,53 @@ def completionEmail(summary_params, email, email_on_fail, plaintext_email, outdi
...
 
     // Render the HTML template
-    def hf            = new File("${workflow.projectDir}/assets/email_template.html")
+    def hf = new File("${workflow.projectDir}/assets/email_template.html")
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:338]:
```suggestion
    def hf            = new File("${workflow.projectDir}/assets/email_template.html")
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 17]
```diff
@@ -325,51 +329,53 @@ def completionEmail(summary_params, email, email_on_fail, plaintext_email, outdi
...
     def html_template = engine.createTemplate(hf).make(email_fields)
-    def email_html    = html_template.toString()
+    def email_html = html_template.toString()
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:340]:
```suggestion
    def email_html    = html_template.toString()
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 18]
```diff
@@ -325,51 +329,53 @@ def completionEmail(summary_params, email, email_on_fail, plaintext_email, outdi
...
     // Render the sendmail template
     def max_multiqc_email_size = (params.containsKey('max_multiqc_email_size') ? params.max_multiqc_email_size : 0) as nextflow.util.MemoryUnit
-    def smail_fields           = [ email: email_address, subject: subject, email_txt: email_txt, email_html: email_html, projectDir: "${workflow.projectDir}", mqcFile: mqc_report, mqcMaxSize: max_multiqc_email_size.toBytes() ]
-    def sf                     = new File("${workflow.projectDir}/assets/sendmail_template.txt")
-    def sendmail_template      = engine.createTemplate(sf).make(smail_fields)
-    def sendmail_html          = sendmail_template.toString()
+    def smail_fields = [email: email_address, subject: subject, email_txt: email_txt, email_html: email_html, projectDir: "${workflow.projectDir}", mqcFile: mqc_report, mqcMaxSize: max_multiqc_email_size.toBytes()]
+    def sf = new File("${workflow.projectDir}/assets/sendmail_template.txt")
+    def sendmail_template = engine.createTemplate(sf).make(smail_fields)
+    def sendmail_html = sendmail_template.toString()
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:347]:
```suggestion
    def smail_fields           = [email: email_address, subject: subject, email_txt: email_txt, email_html: email_html, projectDir: "${workflow.projectDir}", mqcFile: mqc_report, mqcMaxSize: max_multiqc_email_size.toBytes()]
    def sf                     = new File("${workflow.projectDir}/assets/sendmail_template.txt")
    def sendmail_template      = engine.createTemplate(sf).make(smail_fields)
    def sendmail_html          = sendmail_template.toString()
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 19]
```diff
@@ -394,52 +402,61 @@ def completionSummary(monochrome_logs=true) {
...
+    misc_fields['nxf_build'] = workflow.nextflow.build
+    misc_fields['nxf_timestamp'] = workflow.nextflow.timestamp
 
     def msg_fields = [:]
-    msg_fields['version']      = getWorkflowVersion()
-    msg_fields['runName']      = workflow.runName
-    msg_fields['success']      = workflow.success
+    msg_fields['version'] = getWorkflowVersion()
+    msg_fields['runName'] = workflow.runName
+    msg_fields['success'] = workflow.success
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:433]:
```suggestion
    msg_fields['version']      = getWorkflowVersion()
    msg_fields['runName']      = workflow.runName
    msg_fields['success']      = workflow.success
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 20]
```diff
@@ -394,52 +402,61 @@ def completionSummary(monochrome_logs=true) {
...
     msg_fields['dateComplete'] = workflow.complete
-    msg_fields['duration']     = workflow.duration
-    msg_fields['exitStatus']   = workflow.exitStatus
+    msg_fields['duration'] = workflow.duration
+    msg_fields['exitStatus'] = workflow.exitStatus
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:436]:
```suggestion
    msg_fields['duration']     = workflow.duration
    msg_fields['exitStatus']   = workflow.exitStatus
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 21]
```diff
@@ -394,52 +402,61 @@ def completionSummary(monochrome_logs=true) {
...
     msg_fields['errorMessage'] = (workflow.errorMessage ?: 'None')
-    msg_fields['errorReport']  = (workflow.errorReport ?: 'None')
-    msg_fields['commandLine']  = workflow.commandLine.replaceFirst(/ +--hook_url +[^ ]+/, "")
-    msg_fields['projectDir']   = workflow.projectDir
-    msg_fields['summary']      = summary << misc_fields
+    msg_fields['errorReport'] = (workflow.errorReport ?: 'None')
+    msg_fields['commandLine'] = workflow.commandLine.replaceFirst(/ +--hook_url +[^ ]+/, "")
+    msg_fields['projectDir'] = workflow.projectDir
+    msg_fields['summary'] = summary << misc_fields
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:441]:
```suggestion
    msg_fields['errorReport']  = (workflow.errorReport ?: 'None')
    msg_fields['commandLine']  = workflow.commandLine.replaceFirst(/ +--hook_url +[^ ]+/, "")
    msg_fields['projectDir']   = workflow.projectDir
    msg_fields['summary']      = summary << misc_fields
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 22]
```diff
@@ -394,52 +402,61 @@ def completionSummary(monochrome_logs=true) {
...
+    }
+    if (workflow.commitId) {
+        misc_fields['commitid'] = workflow.commitId
+    }
+    if (workflow.revision) {
+        misc_fields['revision'] = workflow.revision
+    }
+    misc_fields['nxf_version'] = workflow.nextflow.version
+    misc_fields['nxf_build'] = workflow.nextflow.build
+    misc_fields['nxf_timestamp'] = workflow.nextflow.timestamp
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:428]:
```suggestion
    misc_fields['start']          = workflow.start
    misc_fields['complete']       = workflow.complete
    misc_fields['scriptfile']     = workflow.scriptFile
    misc_fields['scriptid']       = workflow.scriptId
    if (workflow.repository) {
        misc_fields['repository'] = workflow.repository
    }
    if (workflow.commitId) {
        misc_fields['commitid']   = workflow.commitId
    }
    if (workflow.revision) {
        misc_fields['revision']   = workflow.revision
    }
    misc_fields['nxf_version']    = workflow.nextflow.version
    misc_fields['nxf_build']      = workflow.nextflow.build
    misc_fields['nxf_timestamp']  = workflow.nextflow.timestamp
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 23]
```diff
@@ -394,52 +402,61 @@ def completionSummary(monochrome_logs=true) {
...
+    def engine = new groovy.text.GStringTemplateEngine()
     // Different JSON depending on the service provider
     // Defaults to "Adaptive Cards" (https://adaptivecards.io), except Slack which has its own format
-    def json_path     = hook_url.contains("hooks.slack.com") ? "slackreport.json" : "adaptivecard.json"
-    def hf            = new File("${workflow.projectDir}/assets/${json_path}")
+    def json_path = hook_url.contains("hooks.slack.com") ? "slackreport.json" : "adaptivecard.json"
+    def hf = new File("${workflow.projectDir}/assets/${json_path}")
     def json_template = engine.createTemplate(hf).make(msg_fields)
-    def json_message  = json_template.toString()
+    def json_message = json_template.toString()
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:450]:
```suggestion
    def json_message  = json_template.toString()
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 24]
```diff
@@ -394,52 +402,61 @@ def completionSummary(monochrome_logs=true) {
...
-    msg_fields['projectDir']   = workflow.projectDir
-    msg_fields['summary']      = summary << misc_fields
+    msg_fields['errorReport'] = (workflow.errorReport ?: 'None')
+    msg_fields['commandLine'] = workflow.commandLine.replaceFirst(/ +--hook_url +[^ ]+/, "")
+    msg_fields['projectDir'] = workflow.projectDir
+    msg_fields['summary'] = summary << misc_fields
 
     // Render the JSON template
-    def engine       = new groovy.text.GStringTemplateEngine()
+    def engine = new groovy.text.GStringTemplateEngine()
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:444]:
```suggestion
    def engine       = new groovy.text.GStringTemplateEngine()
```

File: subworkflows/nf-core/utils_nfcore_pipeline/main.nf [hunk 25]
```diff
@@ -394,52 +402,61 @@ def completionSummary(monochrome_logs=true) {
...
     // Different JSON depending on the service provider
     // Defaults to "Adaptive Cards" (https://adaptivecards.io), except Slack which has its own format
-    def json_path     = hook_url.contains("hooks.slack.com") ? "slackreport.json" : "adaptivecard.json"
-    def hf            = new File("${workflow.projectDir}/assets/${json_path}")
+    def json_path = hook_url.contains("hooks.slack.com") ? "slackreport.json" : "adaptivecard.json"
+    def hf = new File("${workflow.projectDir}/assets/${json_path}")
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfcore_pipeline/main.nf:448]:
```suggestion
    def json_path     = hook_url.contains("hooks.slack.com") ? "slackreport.json" : "adaptivecard.json"
    def hf            = new File("${workflow.projectDir}/assets/${json_path}")
```

File: subworkflows/nf-core/utils_nfvalidation_plugin/main.nf [hunk 26]
```diff
@@ -30,30 +29,30 @@ workflow UTILS_NFVALIDATION_PLUGIN {
...
     main:
 
-    log.debug "Using schema file: ${schema_filename}"
+    log.debug("Using schema file: ${schema_filename}")
 
     // Default values for strings
-    pre_help_text    = pre_help_text    ?: ''
-    post_help_text   = post_help_text   ?: ''
+    pre_help_text = pre_help_text ?: ''
+    post_help_text = post_help_text ?: ''
```
Comment (mirpedrol) [inline subworkflows/nf-core/utils_nfvalidation_plugin/main.nf:36]:
```suggestion
    pre_help_text    = pre_help_text    ?: ''
    post_help_text   = post_help_text   ?: ''
```

**Review by mashehu — APPROVED** [review]

**Comment by jfy133** [comment]
Thank you @mirpedrol !

File: subworkflows/nf-core/utils_nextflow_pipeline/main.nf [hunk 27]
```diff
@@ -106,19 +110,13 @@ def checkCondaChannels() {
...
-            "  You will need to set-up the conda-forge and bioconda channels correctly.\n" +
-            "  Please refer to https://bioconda.github.io/\n" +
-            "  The observed channel order is \n" +
-            "  ${channels}\n" +
-            "  but the following channel order is required:\n" +
-            "  ${required_channels_in_order}\n" +
-            "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+        log.warn(
+            "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n" + "  There is a problem with your Conda configuration!\n\n" + "  You will need to set-up the conda-forge and bioconda channels correctly.\n" + "  Please refer to https://bioconda.github.io/\n" + "  The observed channel order is \n" + "  ${channels}\n" + "  but the following channel order is required:\n" + "  ${required_channels_in_order}\n" + "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+        )
```
Comment (bentsherman) [inline subworkflows/nf-core/utils_nextflow_pipeline/main.nf:120]: I tried to preserve the formatting here but found it to be tricky, probably a multi-line string would be better anyway:
```groovy
    log.warn """\
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          There is a problem with your Conda configuration!

          You will need to set-up the conda-forge and bioconda channels correctly.
          Please refer to https://bioconda.github.io/
          The observed channel order is 
          ${channels}
          but the following channel order is required:
          ${required_channels_in_order}
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        """.stripIndent(true)

```

**Pull request merged**